### ISSUE TITLE
Intermittent failure to upgrade charm if new config option is specified

### ISSUE THREAD

**Issue by mthaddon** [juju/terraform-provider-juju#184]
### Terraform Version
```
$ terraform -v
Terraform v1.4.2
on linux_amd64
+ provider registry.terraform.io/juju/juju v0.6.0
```
### Affected Resource(s)
- juju_application

### Terraform Configuration Files
Initial:
```hcl
terraform {
  required_providers {
    juju = {
      source  = "juju/juju"
      version = "0.6.0"
    }
  }
}
provider "juju" {}

resource "juju_application" "githubrunner" {
  name  = "github-runner"
  model = juju_model.githubrunners.name


  charm {
    name     = "github-runner"
    revision = 8
    channel  = "latest/edge"
  }

  units = 1
}```
Then upgrading:
```hcl
terraform {
  required_providers {
    juju = {
      source  = "juju/juju"
      version = "0.6.0"
    }
  }
}
provider "juju" {}

resource "juju_application" "githubrunner" {
  name  = "github-runner"
  model = juju_model.githubrunners.name


  charm {
    name     = "github-runner"
    revision = 10
    channel  = "latest/edge"
  }
  
  config = {
    group = "mygroup"
   }

  units = 1
}
```

### Debug Output

https://paste.ubuntu.com/p/ykYkhQqg8T/

### Panic Output
N/A

### Expected Behavior
It shouldn't have failed the first time.

### Actual Behavior
It failed the first time but then worked the second time

### Steps to Reproduce
See the test above. We're adding a new config option to the charm from revision 8 to 10.

**Comment by juanmanuel-tirado** [comment]
From the provided output I would say this is a problem regarding the order in which we execute the configuration of the charm.
`│ Error: parsing settings for application: unknown option "group"`
Version 8 does not have a `group` configuration option, so probably we're trying to set the option before the new charm is already available. 

Thanks for reporting this one.

**Comment by juanmanuel-tirado** [comment]
This should be fixed in 0.7.0. Reopen if the bug persists.

**Comment by mthaddon** [comment]
We saw this again today with version `0.10.1` of the provider: https://pastebin.ubuntu.com/p/ncp6NBpjT9/
